### ISSUE TITLE
Inconsistent type casting styles found when enclosed in MD5

### ISSUE THREAD

**Issue by matrixik** [quarylabs/sqruff#2062]
### What Happened
Inconsistent type casting styles found when enclosed in MD5. When I remove MD5 it's not showing this warning.

### Expected Behaviour
No error.

### How to reproduce

```sql
CREATE UNIQUE INDEX idx_n ON user_n (user_id, channel_type, MD5(CAST(channel_config AS TEXT)));

```

### Configuration
```ini
[sqruff]
dialect = postgres
large_file_skip_byte_limit = 0
exclude_rules = LT05
output_line_length = 120
max_line_length = 0
rules = all
```

sqruff --version
sqruff 0.29.3

**Comment by matrixik** [comment]
```sql
SELECT CAST(AVG(execution_time_ms) AS INTEGER) AS avg_execution_ms
FROM
    alert_execution_history;

```
Remove AVG and warning disappears.
